lib/if_rmap.c

```diff
@@ -287,18 +297,38 @@ int config_write_if_rmap(struct vty *vty)
 	return write;
 }
 
-void if_rmap_reset(void)
+void if_rmap_ctx_delete(struct if_rmap_ctx *ctx)
+{
+	XFREE(MTYPE_IF_RMAP_CTX, ctx);
+}
+
+struct if_rmap_ctx *if_rmap_ctx_create(struct vrf *vrf)
 {
-	hash_clean(ifrmaphash, (void (*)(void *))if_rmap_free);
+	struct if_rmap_ctx *ctx;
+
+	ctx = XCALLOC(MTYPE_IF_RMAP_CTX, sizeof(struct if_rmap_ctx));
+	ctx->vrf = vrf;
+	ctx->ifrmaphash = hash_create_size(4, if_rmap_hash_make, if_rmap_hash_cmp,
+					   "Interface Route-Map Hash");
+	if (!if_rmap_ctx_list)
+		if_rmap_ctx_list = list_new();
+	listnode_add(if_rmap_ctx_list, ctx);
+	return ctx;
 }
 
 void if_rmap_init(int node)
 {
-	ifrmaphash = hash_create_size(4, if_rmap_hash_make, if_rmap_hash_cmp,
-				      "Interface Route-Map Hash");
 	if (node == RIPNG_NODE) {
 	} else if (node == RIP_NODE) {
 		install_element(RIP_NODE, &if_rmap_cmd);
 		install_element(RIP_NODE, &no_if_rmap_cmd);
 	}
+	if_rmap_ctx_list = list_new();
+}
+
+void if_rmap_destroy(void)
```

Comment:
nit: I think something like `if_rmap_terminate()` would be better in terms of being consistent with the rest of the code base.

Reply:
ok